presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -208,6 +208,18 @@ void PrestoServer::run() {
         clientCertAndKeyPath,
         ciphers);
     announcer_->start();
+    uint64_t heartbeatFrequencyMs = systemConfig->heartbeatFrequencyMs();
```

Comment:
Do we need announcer and heartbeat manager both? I assume when worker is registered with multiple coordinator, worker will heartbeat only to resource manager, not to the coordinator anymore.

Reply:
We need to send both announcement and hearbeats to resource manager.